### PR TITLE
feat(documentation): Updated documentation - OEL-2078

### PR DESCRIPTION
**Pull request by tibi2303**
[skip chromatic]

### CODE REVIEW

File: src/compositions/bcl-file/file.html.twig [hunk 1]
```diff
@@ -21,7 +21,7 @@
   - variant (string) (default: 'default')
     - options: ['default', 'card']
   - download (link Object) (default: {})
-  - translation (object) (default: {}) format: {
+  - translation (translation Object) (default: {}) format: {
```
Comment (drishu) [inline src/compositions/bcl-file/file.html.twig:24]: if we take the name of the template as the base of the "object" name, then it should be:
```suggestion
  - translation (file translation Object) (default: {}) format: {
```

File: src/compositions/bcl-header/header.html.twig [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 {# Parameters:
   - variant (string) (default: 'neutral')
+    - options: ['ec', 'neutral']
```
Comment (drishu) [inline src/compositions/bcl-header/header.html.twig:5]: shouldn't it be the same options as for the footer ?

File: src/compositions/bcl-language-switcher/language-switcher.html.twig [hunk 3]
```diff
@@ -1,9 +1,9 @@
 {% apply spaceless %}
 
 {# Parameters:
-  - id (string) (default: collapse-{random number 0-1000}) - collapse custom id - optional
+  - id (string) (default: collapse-random(1000)) - collapse custom id - optional
```
Comment (drishu) [inline src/compositions/bcl-language-switcher/language-switcher.html.twig:4]:
```suggestion
  - id (string) (default: collapse-random(1000), optional)
```
seemed a bit confusing, lets simplify

File: src/compositions/bcl-listing/listing.html.twig [hunk 4]
```diff
@@ -5,9 +5,9 @@
   - title_tag: (string) (default: 'h2')
   - title_link: (link object) (default: {})
   - title_attributes (drupal attrs)
+  - items: (array of objects of type card)
```
Comment (drishu) [inline src/compositions/bcl-listing/listing.html.twig:8]:
```suggestion
  - items: (card[])
```

File: src/compositions/bcl-recent-activities/recent-activities.html.twig [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {% apply spaceless %}
 
 {# Parameters:
-  - activities (array of custom object) (default: [])
+  - activities (array of object) (default: [])
```
Comment (drishu) [inline src/compositions/bcl-recent-activities/recent-activities.html.twig:4]:
```suggestion
  - activities (object[]) (default: [])
```

**Review by drishu — CHANGES_REQUESTED** [review]
all these "array of this that object", I dont like it, we have to find a good naming convention and then append "[]", like `object[]`

File: src/compositions/bcl-content-banner/content-banner.html.twig [hunk 6]
```diff
@@ -3,18 +3,15 @@
...
+    - title_tag: (string) (default: 'h1')
     - title_link: (link object) (default: {})
     - title_attributes: (drupal attrs)
     - background: (string) (default: 'white')
       options: [ white, gray ]
-    - image (object)
-    - image_size: (string)
-    - date: (object)
-      options: ['sm', 'md', 'lg', 'xl']
-    - content: (string)
```
[the remark this answers is not in the document]
Comment (tibi2303) [reply src/compositions/bcl-content-banner/content-banner.html.twig:15]: done, included the ones that can be used inside the content-banner

File: src/compositions/bcl-content-banner/content-banner.html.twig [hunk 7]
```diff
@@ -3,18 +3,15 @@
...
-    - badges: (array of objects of type badge)
-    - links: (array of objects of type link)
+    - image (object) (default: {})
+    - image_size: (string) (default: 'md')
+      options: ['md', 'lg', 'xl']
+    - date: (object) (default: {})
```
[the remark this answers is not in the document]
Comment (tibi2303) [reply src/compositions/bcl-content-banner/content-banner.html.twig]: changed them to `{component} object` format

File: src/compositions/bcl-file/file.html.twig (hunk 1 above)
Comment (tibi2303) [reply src/compositions/bcl-file/file.html.twig]: changed

File: src/compositions/bcl-listing/listing.html.twig (hunk 4 above)
Comment (tibi2303) [reply src/compositions/bcl-listing/listing.html.twig]: changed

File: src/compositions/bcl-recent-activities/recent-activities.html.twig (hunk 5 above)
Comment (tibi2303) [reply src/compositions/bcl-recent-activities/recent-activities.html.twig]: changed

File: src/compositions/bcl-header/header.html.twig (hunk 2 above)
Comment (drishu) [reply src/compositions/bcl-header/header.html.twig:5]: header takes the bcl-header--neutral, bcl-header--ec and bcl-header--eu classes, at least in bcl 24.1, if this changed revert it, or maybe I'm not understanding something ?

Comment (tibi2303) [reply src/compositions/bcl-header/header.html.twig:5]: ah, you are right, it may confusing because we have the --eu variant as default, it's added

**Review by drishu — APPROVED** [review]

**Pull request merged**